pages/profiles/index.js

```diff
@@ -0,0 +1,99 @@
+// node_modules
+import Link from "next/link"
+import PropTypes from "prop-types"
+// components
+import Breadcrumbs from "../../components/breadcrumbs"
+import PagePreamble from "../../components/page-preamble"
+// libs
+import buildBreadcrumbs from "../../libs/breadcrumbs"
+import Request from "../../libs/request"
+
+const ChildElement = (props) => {
+  const schema = props.schemas[props.schemaKey]
+  if (schema && schema.title) {
+    return (
+      <div className={`px-${props.indentation}`}>
+        <Link href={`${schema["$id"].replace('.json', '')}`} key={props.title}>
+          <a className="block">{props.title}</a>
+        </Link>
+      </div>
+    )
+  } else if (Object.keys(props.child).length > 0) {
+    return (
+      <div className={`px-${props.indentation}`}>
+        <div className="font-semibold">{props.title}</div>
+          {Object.keys(props.child).map((child_key) => {
+            return (
+              <ChildElement
+                title={props.schemas[child_key] ? props.schemas[child_key].title : child_key}
+                schemaKey={child_key}
+                schemas={props.schemas}
+                child={props.child[child_key]}
+                indentation={(props.indentation + 4)}
+                key={child_key}
+              />
```

Comment:
I always love recursive React components; I guess it looks cool (and of course it solves problems).

Reply:
I actually couldn't think of another way to build the hierarchy!